core/src/main/java/com/novoda/merlin/service/ConnectivityChangesForwarder.java

```diff
@@ -0,0 +1,68 @@
+package com.novoda.merlin.service;
+
+import com.novoda.merlin.NetworkStatus;
+import com.novoda.merlin.receiver.ConnectivityChangeEvent;
+import com.novoda.merlin.registerable.bind.BindListener;
+import com.novoda.merlin.registerable.connection.ConnectListener;
+import com.novoda.merlin.registerable.disconnection.DisconnectListener;
+
+class ConnectivityChangesForwarder {
+
+    private final NetworkStatusRetriever networkStatusRetriever;
+    private final DisconnectListener disconnectListener;
+    private final ConnectListener connectListener;
+    private final BindListener bindListener;
+    private final HostPinger hostPinger;
+
+    private NetworkStatus networkStatus;
+
+    ConnectivityChangesForwarder(NetworkStatusRetriever networkStatusRetriever,
+                                 DisconnectListener disconnectListener,
+                                 ConnectListener connectListener,
+                                 BindListener bindListener,
+                                 HostPinger hostPinger) {
+        this.networkStatusRetriever = networkStatusRetriever;
+        this.disconnectListener = disconnectListener;
+        this.connectListener = connectListener;
+        this.bindListener = bindListener;
+        this.hostPinger = hostPinger;
+    }
+
+    void notifyOfInitialNetworkStatus() {
+        if (networkStatus == null) {
+            bindListener.onMerlinBind(networkStatusRetriever.retrieveNetworkStatus());
```

Comment:
I think this would benefit from a renaming 😄  it's never good if someone has to explain it, the code should be self-documenting. I'll see what I can do to improve this.